process/block/pendingMb/pendingMiniBlocks.go

```diff
@@ -61,20 +63,25 @@ func (p *pendingMiniBlocks) getMiniBlocksHashesReadyForCrossShardExecution(metaB
 	return crossShardMiniBlocks, nil
 }
 
-func getMiniBlocksFromHeaderReadyForCrossShardExecution(header data.MetaHeaderHandler) ([]data.MiniBlockHeaderHandler, error) {
-	if !header.IsHeaderV3() {
-		return header.GetMiniBlockHeaderHandlers(), nil
+func (p *pendingMiniBlocks) getMiniBlockHandlersFromShardData(metaBlock data.MetaHeaderHandler) ([]data.MiniBlockHeaderHandler, error) {
+	miniBlocks := make([]data.MiniBlockHeaderHandler, 0)
+	if !metaBlock.IsHeaderV3() {
+		for _, shardData := range metaBlock.GetShardInfoHandlers() {
+			miniblockHandlers := shardData.GetShardMiniBlockHeaderHandlers()
+			miniBlocks = append(miniBlocks, miniblockHandlers...)
+		}
+
+		return miniBlocks, nil
 	}
 
-	miniBlocks := make([]data.MiniBlockHeaderHandler, 0)
-	execResultHandlers := header.GetExecutionResultsHandlers()
-	for _, execResult := range execResultHandlers {
-		mbs, errGetMbs := common.GetMiniBlocksHeaderHandlersFromExecResult(execResult)
-		if errGetMbs != nil {
-			return nil, errGetMbs
+	shardDataProposalHandlers := metaBlock.GetShardInfoProposalHandlers()
+	for _, shardData := range shardDataProposalHandlers {
```

Comment:
Slightly more efficient
```suggestion
	for _, shardData := range metaBlock.GetShardInfoProposalHandlers() {
```